### ISSUE TITLE
When running on CI/CD through pre-commit it cannot load installed collections

### ISSUE THREAD

**Issue by aarroba** [ansible/ansible-lint#3865]
##### Summary

I have a Gitlab pipeline that runs ansible-lint through pre-commit configuration and always fails loading/finding ansible.windows.win_file. Same pre-commit config works locally. Tested installing ansible and the failing collection on the before_script configuration of the pipelina

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

python:3.11 docker image

- ansible installation method: pre-commit or pip
- ansible-lint installation method: pre-commit

##### STEPS TO REPRODUCE

```yaml
#gitlab-ci.yaml
---
workflow:
  rules:
    # Allow running merge request pipeline when a merge request is open with current branch as source.
    - if: $CI_PIPELINE_SOURCE == "merge_request_event"
    # Disallow running branch pipeline when there is an open merge request from current branch.
    - if: $CI_COMMIT_BRANCH && $CI_OPEN_MERGE_REQUESTS
      when: never
    # If no merge request is open and there are changes in the branch, run a branch pipeline.
    - if: $CI_COMMIT_BRANCH
      changes:
        compare_to: "refs/heads/main"
        paths:
          - "*"

stages:
  - lint

precommit:
  stage: lint
  image: python:3.11
  variables:
    PRE_COMMIT_HOME: ${CI_PROJECT_DIR}/.cache/pre-commit
  cache:
    paths:
      - ${PRE_COMMIT_HOME}
  before_script:
    - pip install -q pre-commit
    - pip install ansible
    - ansible-galaxy collection install ansible.windows community.windows
  script:
    - pre-commit run --show-diff-on-failure --color=always --all-files

```

```yaml
# pre-commit-config.yml
# See https://pre-commit.com for more information
# See https://pre-commit.com/hooks.html for more hooks
repos:
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.4.0
    hooks:
      - id: trailing-whitespace
      - id: end-of-file-fixer
      - id: check-yaml
      - id: check-added-large-files
      - id: check-case-conflict

  - repo: https://github.com/ansible/ansible-lint
    rev: v6.21.1 # Replace this with the wished ansible-lint version
    hooks:
      - id: ansible-lint
        stages:
          - commit

```

```yaml
# contadoresiis.yml
---
- name: Configuracion de contadores de IIS
  hosts: "{{ cliente | default('Null') }}"
  gather_facts: false
  roles:
    - contadoresiis

```

```yaml
# roles/contadoresiis/tasks/main.yml
---
- name: Create folders
  ansible.windows.win_file:
    path: C:\whatever
    state: directory
  failed_when: false

- name: Copy folder
  ansible.windows.win_copy:
    src: files/
    dest: "C:/whatever"
    force: true
  failed_when: false

- name: Run task to schedule the exporting data
  ansible.windows.win_command: "some command"

- name: Restart service
  ansible.windows.win_service:
    name: xxxxx
    state: restarted

```

```yaml
# .ansible-lint
skip_list:
  - latest
  - var-naming[no-role-prefix]
  - name[template]
  - no-changed-when

```

##### Desired Behavior

ansible-lint ends successfully

##### Actual Behavior

ansible-lint is unable to load an installed module.

Relevant output of the pipeline:
```console
Starting galaxy collection install process
Nothing to do. All requested collections are already installed. If you want to reinstall them, consider using `--force`.
$ pre-commit run --show-diff-on-failure --color=always --all-files
[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/ansible/ansible-lint.
[INFO] Initializing environment for https://github.com/ansible/ansible-lint:ansible-core>=2.13.3.
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/ansible/ansible-lint.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
trim trailing whitespace.................................................Passed
fix end of files.........................................................Passed
check yaml...............................................................Passed
check for added large files..............................................Passed
check for case conflicts.................................................Passed
Ansible-lint.............................................................Failed
- hook id: ansible-lint
- exit code: 2
INFO     Identified /builds/Comun/Monitorizacion/smart-monitoring/Install_Telegraf as project root due .git directory.
INFO     Set ANSIBLE_LIBRARY=/root/.cache/ansible-compat/22809f/modules:/root/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/root/.cache/ansible-compat/22809f/collections:/root/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/root/.cache/ansible-compat/22809f/roles:roles:/root/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
INFO     Loading ignores from .gitignore
INFO     Excluded: .cache
INFO     Excluded: .git
INFO     Loading ignores from .gitignore
INFO     Excluded: .cache
INFO     Excluded: .git
INFO     Set ANSIBLE_LIBRARY=/root/.cache/ansible-compat/22809f/modules:/root/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/root/.cache/ansible-compat/22809f/collections:/root/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/root/.cache/ansible-compat/22809f/roles:roles:/root/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
INFO     Executing syntax check on role roles/regdef_keys (0.72s)
INFO     Executing syntax check on role roles/telegraf (0.73s)
INFO     Executing syntax check on playbook contadores.yml (0.69s)
INFO     Executing syntax check on playbook site.yml (0.85s)
INFO     Executing syntax check on role roles/contadoresiis (0.66s)
WARNING  Listing 1 violation(s) that are fatal
d=585752;https://ansible-lint.readthedocs.io/rules/syntax-check/1;91msyntax-check[specific]: couldn't resolve module/action 'ansible.windows.win_file'. This often indicates a misspelling, missing collection, or incorrect module path.
roles/contadoresiis/tasks/main.yml:2:3
                  Rule Violation Summary                   
 count tag                    profile rule associated tags 
     1 d=96869;https://ansible-lint.readthedocs.io/rules/syntax-check31msyntax-check[specific] min     core, unskippable    
Failed: 1 failure(s), 0 warning(s) on 32 files.
ERROR: Job failed: exit code 1
```
It doesn't matter if the installation of Ansible is made on before_script or handled by pre-commit.

Thank you so much!
